service/src/main/java/com/codeforcommunity/requester/S3Requester.java

```diff
@@ -273,4 +282,48 @@ public static String uploadHTML(
 
     return String.format("%s/%s/%s", externs.getBucketPublicUrl(), directoryName, name);
   }
+
+  // helper to check whether the given path exists
+  public static boolean pathExists(String path) {
+    Boolean exists = externs.getS3Client().doesObjectExist(externs.getBucketPublic(), path);
+    return exists;
+  }
+
+  /**
+   * Load the existing HTML file with the given name from the user uploads S3 bucket.
+   *
+   * @param name the name of the HTML file in S3 to be loaded (without "_template.html extension).
+   * @param directoryName the directory of the file in S3 (without leading or trailing '/').
+   * @return LoadTemplateResponse with the html file name, content, and author.
+   * @throws InvalidURLException if the file does not exist.
+   * @throws BadRequestHTMLException if the HTML file cannot be decoded to a string.
+   * @throws SdkClientException if the loading from S3 failed.
+   */
+  public static LoadTemplateResponse loadHTML(String name, String directoryName) {
+    String HTMLPath = directoryName + "/" + name + "_template.html";
+
+    if (!pathExists(HTMLPath)) {
+      throw new InvalidURLException();
+    }
+
+    // Create the request to delete the HTML
+    GetObjectRequest awsRequest = new GetObjectRequest(externs.getBucketPublic(), HTMLPath);
+
+    S3Object HTMLFile = externs.getS3Client().getObject(awsRequest);
+    StringBuilder content = new StringBuilder();
+    try {
+      int c = 0;
+      while ((c = HTMLFile.getObjectContent().read()) != -1) {
+        content.append((char) c);
+      }
+    } catch(IOException e) {
+      throw new BadRequestHTMLException("HTML file could not be decoded to string");
+    }
+
+    String HTMLContent = content.toString();
+
+    String HTMLAuthor = HTMLFile.getObjectMetadata().getUserMetaDataOf("userID");
```

Comment:
:pick: Can we make `HTMLAuthor` the uploader's first (or full) name so that it'll be more recognizable by the admin in the frontend? But if we don't need to show their name, we might not need the author in the response. Also, even more of a nitpick but we can rename `HTMLAuthor` and the other variables to `htmlAuthor`, etc.

Reply:
I did the db querying in the processor impl class to get the first and last name. It is a little hacky in that I am isntantiating the LoadTemplateResponse dto twice (once from the S3 requester with the id as the author, and once after querying with the full name as the author to finally return).